pkg/cmd/roachprod-microbench/clean.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+type cleanConfig struct {
+	inputFilePath  string
+	outputFilePath string
+}
+
+type clean struct {
+	cleanConfig
+	inputFile *os.File
+
+	ctx context.Context
+}
+
+func defaultCleanConfig() cleanConfig {
+	return cleanConfig{
+		inputFilePath:  "",
+		outputFilePath: "",
+	}
+}
+
+func newClean(config cleanConfig) (*clean, error) {
+	file, err := os.Open(config.inputFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := context.Background()
+
+	return &clean{cleanConfig: config, inputFile: file, ctx: ctx}, nil
+}
+
+func (c *clean) writeCleanOutputToFile(cleanedBenchmarkOutputLog benchmarkExtractionResult) error {
+
+	if err := os.MkdirAll(c.outputFilePath[:strings.LastIndex(c.outputFilePath, "/")], os.ModePerm); err != nil {
```

Comment:
I think you can use `https://pkg.go.dev/path/filepath#Dir` here to give you only the directories.